ilastik/workflows/carving/carvingGui.py

```diff
@@ -793,23 +793,23 @@ def onButtonsEnabled(slot, roi):
         return layers
 
 
-def _str_to_int(data_str: str) -> Union[str, int]:
-    """
-    Convert string to int when possible
-    """
-    if data_str.isdigit():
-        return int(data_str)
-    return data_str
-
-
 def _humansort_key(elem: str):
     """
-    Key for human sort
-    >>> lst = ['a 1', 'b 2', 'a 10', 'a 9']
+    If there is a trailing number, sort it numerically and not alphabetically
+    >>> lst = ['3', 'c', 'a 1', 'b 2', 'a 10', 'a 9']
+    >>> sorted(lst)
+    ['3','a 1', 'a 10', 'a 9', 'b 2', 'c']
     >>> sorted(lst, key=_humansort_key)
-    ['a 1', 'a 9', 'a 10', 'b 2']
+    ['3','a 1', 'a 9', 'a 10', 'b 2', 'c']
     """
     if not (elem and isinstance(elem, str)):
         return tuple()
-
-    return tuple(_str_to_int(token) for token in re.split("(\d+)", elem) if token)
+    split = [token for token in re.split(r"(\d+)$", elem) if token]
```

Comment:
Awesome!

However, this will give (somewhat) strange result on `["x1", "x0", "x"]` -> `["x0", "x", "x1"]` because `x0` and `x` map to the same key, and Python sort is stable. It seems more reasonable to sort this input as `["x", "x0", "x1"]`.

Instead of explicitly converting to int, we could try to sort numeric suffixes as strings too, but compare their lengths first.

Also, I think it's better to precompute all keys since this key func is not just data projection (probably won't make a huge difference though).

```python
def humansorted(items):
    digits = "0123456789"
    keys = {}
    for item in items:
        i = next((i for i in range(len(item), 0, -1) if item[i - 1] not in digits), 0)
        keys[item] = item[:i], -i, item[i:]
    return sorted(items, key=keys.__getitem__)
```

Reply:
Totally forgot about the `strip` methods! Obviously, they are much better.